File: 102.binary-tree-level-order-traversal.py
# @leet start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root: return []
        answer = []
        queue = [root]
        while len(queue) > 0:
            list = []
            length = len(queue)
            for i in range(0, length):
                node = queue.pop(0)
                if node.left is not None:
                    queue.append(node.left)
                if node.right is not None:
                    queue.append(node.right)
                list.append(node.val)
            answer.append(list)
        return answer


# @leet end

File: 102.binary-tree-level-order-traversal.java
// @leet start

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
  public List<List<Integer>> levelOrder(TreeNode root) {
    if (root == null) {
      return new ArrayList<>(); // Return an empty list if the root is null
    }
    Queue<TreeNode> queue = new LinkedList<>();
    queue.offer(root);
    List<List<Integer>> answer = new ArrayList<>();
		

    while (!queue.isEmpty()) {
      int length = queue.size();
      List<Integer> row = new ArrayList<>();
      for (int i = 0; i < length; i++) {
        TreeNode node = queue.remove();
        row.add(node.val);
        if (node.left != null) {
          queue.offer(node.left);
        }
        if (node.right != null) {
          queue.offer(node.right);
        }
      }
      answer.add(row);
    }
    return answer;
  }
}
// @leet end
